x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -272,6 +272,7 @@ function initializeExecutionStatus(
         status: 'pending',
         lastExecutionDate: new Date().toISOString(),
         error: null,
+        warning: null,
```

Comment:
Not sure about this change... don't want to change anything in migrations, but since warning field is not optional i have to either set it null or cast the parameter as RawRule... open for any recommendation here.

Reply:
I'm also leaning on excluding this change. I'm thinking this code was needed to make TypeScript happy? If so, I think at some point, preferably now, we'll have to move away from `RawRule` and go with something decoupled from the latest definition. As we add further changes to the rule SO in the future, the migration code and the up-to-date version will diverge more and more.

Reply:
Ya, in general we don't need migration for properties that we allow to be null-ish in the new code.  But we need to be prepared to get an `undefined` back as the value vs a `null`, if we don't explicitly migrate to `null`.  Haven't reviewed this PR yet though, so not sure of what the effect is.